astropy/cosmology/io/table.py

```diff
@@ -110,7 +113,11 @@ def from_table(table, index=None, *, move_to_meta=False, cosmology=None):
     # string index uses the indexed column on the table to find the row index.
     if isinstance(index, str):
         if not table.indices:  # no indexing column, find by string match
-            indices = np.where(table["name"] == index)[0]
+            nc = "name"  # default name column
+            if rename is not None:  # from inverted `rename`
+                nc = {v: k for k, v in rename.items()}.get(nc, nc)
```

Comment:
I don't quite like constructing a temporary `dict` here.
```suggestion
                for key, value in rename.items():
                    if value == "name":
                        nc = key
                        break
```